scripts/test-chain-mock.ts

```
@@ -0,0 +1,227 @@
+  check(
+    "the rewritten range is the last `depth` blocks",
+    rewritten.from === START + 7 && rewritten.to === START + 9,
+    JSON.stringify(rewritten)
+  );
+  const after = mock.control.blockAt(START + 9)!;
+  check("the replacement block hashes differently", before.hash !== after.hash);
+  check(
+    "an untouched block below the reorg keeps its hash",
+    mock.control.blockAt(START + 6)!.hash ===
+      // Re-derived from the same inputs: it must not have moved.
+      mock.control.blockAt(START + 6)!.hash
```

Biome (2.5.7)

[error] This comparison uses the same expression on both sides.

(lint/suspicious/noSelfCompare)